### DISCUSSION TITLE
Error with passing in non default prompts.

### DISCUSSION THREAD

**Discussion by TusharA912** [sinaptik-ai/pandas-ai#367] [Q&A] [answered]
Hi, I am having a hard time figuring out how to pass in nondefault prompts with pandasai. I am using a dictionary to set up my new prompt as a prompt object from pandas: 
`newprompt = GeneratePythonCodePrompt()`
`pandas_ai = PandasAI(llm=llm, non_default_prompts={"generate_python_code": newprompt})`

But when I run this like so:
`response = pandas_ai.run(df, user_question, is_conversational_answer=True, show_code=True)`

 I get the error: 
"Unfortunately, I was not able to answer your question, because of the following error:\n\n'GeneratePythonCodePrompt' object is not callable\n"

Am I setting it up wrong? I would love an example to figure out how to do this properly

**Comment by gventuri** [comment, marked as answer]
Hi @TusharA912, you should rather pass the class, not the object.

Try this:

```
pandas_ai = PandasAI(llm=llm, non_default_prompts={"generate_python_code": GeneratePythonCodePrompt})
```
